Vb code cleanup 3 of 4

paul1956 opened the pull request:
<!-- Please read CONTRIBUTING.md before submitting a pull request -->

Fixes #


## Proposed changes
Clean up Visual Basic Code to eliminate suggestions, fixes 85 issues flagged by Visual Studio

<!-- We are in TELL-MODE the following section must be completed -->

## Customer Impact
None, if targets developers who get 85 issues that clutter up the Error List

## Regression? 
 No, it more about making code match .editorConfig included with project. Alternativity, .editorConfig could be changed to match the Visual Basic Code but that will cause other cosmetic issues, where other code does not match newer .editorConfig.

## Risk
Very low, mostly renaming of local variables, fix spelling errors in comments and adding missing "Public" on public functions, biggest change is If Statement to if Expression.

<!-- end TELL-MODE -->

## Test methodology
Code completes, existing tests pass, changes are mostly cosmetic.

This is only in 1 Project written in old Visual Basic


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10117)

lonitra reviewed (commented):
LGTM 👍 but will also let @KlausLoeffelmann have a look as well

KlausLoeffelmann reviewed (changes requested):
Hey @paul1956,

I browsed about half through it, but I have a few things generically, which I would either suggest addressing or have a conversation about. Can you take a 2nd look?

Thanks so much for doing this - a few of those things were bothering me also for a long time!

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/AssemblyInfo.vb:
```diff
@@ -40,11 +41,10 @@ Namespace Microsoft.VisualBasic.ApplicationServices
-                    Else
-                        _description = Attribute.Description
-                    End If
+                    _description = If(Attribute Is Nothing,
```
@KlausLoeffelmann Because whoever set up .editorConfig for selected prefer If Expressions. I agree it is not as clear in VB as C# but changing EditorConfig to prefer If Statements might cause bigger changes. It's possible for VB and C# to have different preference for the two languages but for now they are the same.

paul1956 replied:
@KlausLoeffelmann 
I believe I have undone the changes you didn't like and updated editorConfig to separate VB and C# for the three places that were causing this.

VB Now prefers If Statements to Coalesce, If Statements to If Expressions and If Statements to Return Expressions but left C# unchanged. 

The current WinForms Repo is missing a lot of standard options and if editorConfig is edited in Visual Studio through the UI they will be automatically added not necessarily with the desired values. If this needs to be fixed, it should be a separate PR as it probably affects C#.

paul1956 replied:
> > and if editorConfig is edited in Visual Studio through the UI they will be automatically added not necessarily with the desired values.
> 
> Couldn't we just have the missing entries being filled in and then edited (only for VB of course) in the way, we want it? Since VB seems to be behaving in other areas fine, I would figure that the default behavior of the VB Editor _without_ .editorConfig and the default values, if VS fills them in, should be about the same? Or is that a wrong assumption?

@KlausLoeffelmann that should be a correct assumption, but I would not want to make a change that impacts formatting/issues in C#. Right now, the Cleanup PR has a modified editorConfig that gives you VB without suggestions and C# unchanged. There may be additional lines you should like to add but they only effect C#. You can see them by opening editorConfig by double clicking and then saving, then manually review the new settings. I don't know for example how you want collection initialization handled in C#.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Audio.vb:
```diff
@@ -153,13 +157,15 @@ Namespace Microsoft.VisualBasic
             Private Shared Sub ValidateAudioPlayModeEnum(value As AudioPlayMode, paramName As String)
                 If value < AudioPlayMode.WaitToComplete OrElse value > AudioPlayMode.BackgroundLoop Then
-                    Throw New ComponentModel.InvalidEnumArgumentException(paramName, DirectCast(value, Integer), GetType(AudioPlayMode))
+                    Throw New ComponentModel.InvalidEnumArgumentException(paramName, value, GetType(AudioPlayMode))
```
It's not an implicit conversion VB knows AudioPlayMode is an integer because its private and CInt or DIrectCast causes a suggestion.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Audio.vb:
```diff
@@ -153,13 +157,15 @@ Namespace Microsoft.VisualBasic
             Private _sound As Media.SoundPlayer
 
         End Class 'Audio
+
```
The auto formatting is doing this on Save. I could revert using Notepad.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/ComputerInfo.vb:
```diff
@@ -17,7 +17,7 @@ Namespace Microsoft.VisualBasic.Devices
         ''' Default constructor
         ''' </summary>
-        Sub New()
+        Public Sub New()
```
Yes, if I changed to the wrong accessibility, you get warning below

Warning	RS0017	Symbol 'Microsoft.VisualBasic.Devices.ComputerInfo.New() -> Void' is part of the declared API, but is either not public or could not be found.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:
```diff
@@ -3,15 +3,18 @@
 Imports System.Net
 Imports System.Security
 Imports System.Threading
+
```
They are added by automatic former on save.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:
```diff
@@ -129,7 +133,7 @@ Namespace Microsoft.VisualBasic.Devices
             ' However, it is good practice to verify address before calling address.Host.
             If address Is Nothing Then
-                Throw ExceptionUtils.GetArgumentNullException("address")
+                Throw GetArgumentNullException("address")
```
Leaving it cause suggestion.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:
```diff
@@ -144,15 +148,12 @@ Namespace Microsoft.VisualBasic.Devices
-                Return True
-            End If
-            Return False
+            Return Reply.Status = NetInfoAlias.IPStatus.Success
```
I reverted but this should also be set in  editorConfig

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:
```diff
@@ -275,7 +276,7 @@ Namespace Microsoft.VisualBasic.Devices
         ''' <param name="connectionTimeout">Time alloted before giving up on a connection</param>
         ''' <param name="overwrite">Indicates whether or not the file should be overwritten if local file already exists</param>
-        Sub DownloadFile(address As Uri,
+        Public Sub DownloadFile(address As Uri,
```
Same issue its defined as Public in PublicAPI.Shipped as Public

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:
```diff
@@ -674,7 +675,7 @@ Namespace Microsoft.VisualBasic.Devices
                     For i As Integer = 0 To BUFFER_SIZE - 1
                         'This is the same logic Ping.exe uses to fill it's buffer
-                        _pingBuffer(i) = System.Convert.ToByte(Asc("a"c) + i Mod 23, System.Globalization.CultureInfo.InvariantCulture)
+                        _pingBuffer(i) = Convert.ToByte(Asc("a"c) + i Mod 23, Globalization.CultureInfo.InvariantCulture)
```
I thought the same thing and VB Optimizes CByte(Asc since VB 16.0. I will change and test should pick up any issues.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:
```diff
@@ -713,11 +714,10 @@ Namespace Microsoft.VisualBasic.Devices
-            End If
-
-            Return New NetworkCredential(userName, password)
+            Return If(userName.Length = 0 And password.Length = 0,
```
Reverted but again should be changed on editorConfig under VB and specify C# preferences under C# instead of globally.

# VB files
[*.vb]
dotnet_style_prefer_conditional_expression_over_assignment = false:none
dotnet_style_prefer_conditional_expression_over_return = false:none

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:
```diff
@@ -806,15 +806,18 @@ Namespace Microsoft.VisualBasic.Devices
 
         ' Flag used to indicate whether or not we should use passive mode when ftp downloading
         Private _useNonPassiveFtp As Boolean
+
```
This is how I typically do it. Let me know if you want me to change

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Helpers/Hosting.vb:
```diff
@@ -5,16 +5,17 @@ Namespace Microsoft.VisualBasic.CompilerServices
 
     <ComponentModel.EditorBrowsable(ComponentModel.EditorBrowsableState.Never)>
     Public Interface IVbHost
+
```
The whole extra line thing is cause by Saving VB Code and autoformatting.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Helpers/NativeTypes.vb:
```diff
@@ -155,13 +160,15 @@ Namespace Microsoft.VisualBasic.CompilerServices
 
         ' GetWindow() Constants
         Friend Const GW_HWNDFIRST As Integer = 0
+
```
@KlausLoeffelmann If you are asking about all the blank lines all I did was ^K^D Save and the VB formatter does it based on .editorConfig settings.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Helpers/NativeTypes.vb:
```diff
@@ -52,13 +55,14 @@ Namespace Microsoft.VisualBasic.CompilerServices
             Protected Overrides Function ReleaseHandle() As Boolean
                 Return NativeMethods.CloseHandle(handle) <> 0
             End Function
+
```
You don't have to introduce anything, there is already a VB section in the top one in WinForms but there is very little in it. By default, it uses the common stuff in this section which as you correctly noticed is not optimal for VB.

```
# C# and VB files
[*.{cs,vb}]
```

C# then has its own section starting with
```
# C# files
[*.cs]
```

Lastly VB has a dedicated section starting with
```
[*.vb]
```
But it only has 1 line
```
# Modifier preferences
visual_basic_preferred_modifier_order = Partial,Default,Private,Protected,Public,Friend,NotOverridable,Overridable,MustOverride,Overloads,Overrides,MustInherit,NotInheritable,Static,Shared,Shadows,ReadOnly,WriteOnly,Dim,Const,WithEvents,Widening,Narrowing,Custom,Async:suggestion
```
BTW no idea why there is more than one .editorConfig in WinForms repo but there are dozens of them, and I have no idea which is causing the suggestions because editing the TOP one seems to do nothing

KlausLoeffelmann commented:
I would be surprised, if the Roslyn team doesn't have a dedicated .editorConfig for VB. Couldn't we steal that from them? I cannot believe that the current setting we're using should be some sort of standard for VB. 

@AlekseyTs: Asking for a friend! 😸

AlekseyTs commented:
Sorry. I prefer to stay oblivious of the .editorConfig stuff. Feel free to check what we have in the Roslyn repo. I personally never reformat entire file, and I would revert spacing changes if VS were to make them without me explicitly looking to make the changes. Regardless of what .editorConfig has.

paul1956 commented:
> I would be surprised, if the Roslyn team doesn't have a dedicated .editorConfig for VB. Couldn't we steal that from them? I cannot believe that the current setting we're using should be some sort of standard for VB.
> 
> @AlekseyTs: Asking for a friend! 😸

Roslyn has 1 editorConfig for the solution (VB and C#), a few ancillary repos have their own which inherit from the root one.

Spacing is one issue where blank lines after classes, punctions is currently specified and saving applies the formatting. it's possible there is a VS Setting to prevent this the auto formating but I haven't found it.

A separate issue is preferring If Expressions for C# and If Statements for VB.  This is controlled by .editorConfig and right now they share the same settings to prefer If Expressions.

paul1956 commented:
@KlausLoeffelmann I have manually removed all the extra lines that VS wants to add, the biggest manual change is separating Imports but root type. It that is a setting I don't know how to turn it off.

I have added ExceptionUtils and other prefixes like it back. The code was not consistent in how/when it used ExceptionUtils directly vs in an Imports. I have made all uses follow below
```
Imports ExUtils = Microsoft.VisualBasic.CompilerServices.ExceptionUtils
```

paul1956 commented:
@KlausLoeffelmann  Ping
This is all cleaned up with updated and separated editorConfig and updated VB project file with options specified. I think I removed all the extra blank lines, but any editing in VS will bring them back. Someone that knows about controlling the blank lines should review and recommend a fix that is more automatic and not requiring removing the blanks using Notepad. This should be reviewed and merged before the Fix to remove WebClient which has all the same formatting issues.

KlausLoeffelmann commented:
I need to take a bit more time, which I don't have right now.
I'll dial back to this in a bit. Sorry for the delay!

paul1956 commented:
> I need to take a bit more time, which I don't have right now. I'll dial back to this in a bit. Sorry for the delay!

@KlausLoeffelmann no issue, I continue to fix formatting differences between different files. I will continue to keep this PR up to date with Master.

KlausLoeffelmann commented:
When you get the chance, can you also start using the name "main" for the equivalence of our main branch to avoid confusion? Thx!

lonitra commented:
@KlausLoeffelmann gentle ping to keep this on your radar to take this to completion

KlausLoeffelmann commented:
Yes, Ma'am. 
As soon as I can get to it! Give me a bit more time - thanks for pinging me!

paul1956 commented:
Ping, is there an ETA when someone might review this?

paul1956 commented:
This is replace by other PR's and is unneccessary
